Marlin/src/inc/SanityCheck.h

```diff
@@ -4282,6 +4282,10 @@ static_assert(_PLUS_TEST(4), "HOMING_FEEDRATE_MM_M values must be positive.");
   #endif
 #endif
 
+#if ENABLED(BLTOUCH_VALIDATE_ON_HOMING) && DISABLED(Z_SAFE_HOMING)
+  #error "BLTOUCH_VALIDATE_ON_HOMING requires Z_SAFE_HOMING."
```

Comment:
`Z_SAFE_HOMING` makes sense for most users homing with a probe, but if probe/BLTouch is on the right side of a hotend/X carriage, you can usually forgo`Z_SAFE_HOMING` since it's over the bed when X is homed.

Reply:
Updated it to not rely on `Z_SAFE_HOMING`!